server.go

```diff
@@ -1258,13 +1266,18 @@ func (sp *serverPeer) OnGetBlocks(_ *peer.Peer, msg *wire.MsgGetBlocks) {
 
 // OnGetHeaders is invoked when a peer receives a getheaders wire message.
 func (sp *serverPeer) OnGetHeaders(_ *peer.Peer, msg *wire.MsgGetHeaders) {
-	// Ignore getheaders requests if not in sync unless the local best chain
-	// is exactly at the same tip as the requesting peer.
-	locatorHashes := msg.BlockLocatorHashes
+	// Send an empty headers message in the case the local best known chain does
+	// not have the minimum cumulative work value already known to have been
+	// achieved on the network.  This signals to the remote peer that there are
+	// not any interesting headers available without appearing unresponsive.
 	chain := sp.server.chain
-	if !sp.server.syncManager.IsCurrent() && (len(locatorHashes) == 0 ||
-		*locatorHashes[0] != chain.BestSnapshot().PrevHash) {
-
+	tipHash := chain.BestSnapshot().Hash
+	workSum, err := chain.ChainWork(&tipHash)
+	if err == nil && workSum.Lt(&sp.server.minKnownWork) {
+		srvrLog.Debugf("Sending empty headers to peer %s in response to "+
+			"getheaders due to local best knwon tip having too little work",
+			sp.Peer)
```

Comment:
nit: Is the `%s` directive appropriate for `sp.Peer`? Maybe you mean `sp.Peer.String()`? or use the `%v` directive.

Reply:
Yep, `%s` is intended as it is the specific formatting directive for strings and `sp.Peer` implements [fmt.Stringer](https://pkg.go.dev/fmt#Stringer).